### PR TITLE
Record XLA metrics in Tensorboard Summary

### PR DESCRIPTION
**Pull request by zcain117**
These will be picked up by our metrics regressions test, which will alert if any XLA metrics are outside their typical range.

### CODE REVIEW

**Comment by zcain117** [comment]
@dlibenzi   Right now I'm recording the sum of aten ops rather than specific counts of each op. This is because the metric regression test will not alert the first time it sees a new metric; it will wait until it sees N instances of that metric so that we can compute typical bounds. If there were no aten ops and suddenly one day we had new aten ops, then with this strategy we will have a record of 0 aten ops per day and an alert will fire since we are now above 0. If we instead record specific counts of each op, then no alert will fire since it will be the first time seeing that aten op as a metric.

We can also have the special case logic in the metrics handler that reads these Tensorboard summaries, but I'd like to avoid that code needing to know about XLA metrics and instead just be a generic module that reacts to metrics without knowing what each metric means

File: test/test_train_cifar.py [hunk 1]
```diff
@@ -231,6 +231,7 @@ def test_loop_fn(model, loader, device, context):
     global_step = (epoch - 1) * num_training_steps_per_epoch
     test_utils.add_scalar_to_summary(writer, 'Accuracy/test', accuracy,
```
[the remark this answers is not in the document]
Comment (zcain117) [reply test/test_train_cifar.py]: Switched to unified API and re-ran the tests locally to make sure it works

File: torch_xla/test/test_utils.py [hunk 2]
```diff
@@ -10,19 +12,33 @@ def _get_device_spec(device):
...
+                     write_xla_metrics=False):
+  """Writes scalars to a SummaryWriter. Optionally writes XLA perf metrics.
 
   Args:
     summary_writer: instance of Tensorboard SummaryWriter or
         None. If None, no summary files will be written.
-    metric_name: Str. Name of the metric to write.
-    metric_value: Float. Value of the metric.
-    global_step: Int. The global step value for this data point.
+    global_step: Int. The global step value for these data points.
```
Comment (dlibenzi) [inline torch_xla/test/test_utils.py]: This is not the syntax we use. PTAL at examples in xla_model.py

File: torch_xla/test/test_utils.py [hunk 3]
```diff
@@ -10,19 +12,33 @@ def _get_device_spec(device):
...
+  if summary_writer is None:
+    return
+  for k, v in dict_to_write.items():
+    summary_writer.add_scalar(k, v, global_step)
+
+  if write_xla_metrics:
+    metrics = mcu.parse_metrics_report(met.metrics_report())
+    aten_ops_sum = 0
+    for metric_name, metric_value in metrics.items():
+      if metric_name.find("aten::") == 0:
```
Comment (dlibenzi) [inline torch_xla/test/test_utils.py]: Hmm, I think it is OK to have a global count, but these `aten::` metrics should be also written individually.

File: torch_xla/test/test_utils.py (hunk 2 above)
Comment (zcain117) [reply torch_xla/test/test_utils.py]: Fixed. I'll fix the other methods in this file in a separate cleanup PR

File: torch_xla/test/test_utils.py (hunk 3 above)
Comment (zcain117) [reply torch_xla/test/test_utils.py]: Done

File: torch_xla/test/test_utils.py (hunk 2 above)
Comment (zcain117) [reply torch_xla/test/test_utils.py]: Filed a bug to remember to do that

**Comment by zcain117** [comment]
> Be sure the TB summary writer like the colons in the `aten::*` metrics ...

So far I haven't seen any issues with either aten:: or xla::  in the SummaryWriter but good to be mindful of that

**Pull request merged**